hudi-platform-service/hudi-metaserver/hudi-metaserver-client/src/main/java/org/apache/hudi/metaserver/client/HoodieMetaserverClientImp.java

```diff
@@ -61,7 +62,12 @@ public HoodieMetaserverClientImp(HoodieMetaserverConfig config) {
     this.retryDelaySeconds = config.getConnectionRetryDelay();
     String uri = config.getMetaserverUris();
     if (isLocalEmbeddedMetaserver(uri)) {
-      this.client = HoodieMetaserver.getEmbeddedMetaserver();
+      try {
+        Method method = Class.forName("org.apache.hudi.metaserver.HoodieMetaserver").getMethod("getEmbeddedMetaserver", new Class[]{});
+        this.client = (ThriftHoodieMetaserver.Iface) method.invoke(null, new Object[]{});
+      } catch (ClassNotFoundException | NoSuchMethodException | InvocationTargetException | IllegalAccessException e) {
+        throw new HoodieException("Please check the server uri has ever been set. Empty uri is used for local unit test", e);
+      }
```

Comment:
see how can we move this into a helper in org.apache.hudi.common.util.ReflectionUtils so we don't have so many exceptions to catch

Reply:
fix